src/main/webapp/app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component.ts

```diff
@@ -35,7 +35,11 @@ export class ProgressBarComponent implements OnChanges, OnDestroy {
 
     ngOnChanges(changes: SimpleChanges) {
         if (changes.percentage) {
-            this.percentage = round(this.percentage);
+            if (isNaN(this.percentage) && this.numerator === 0 && this.denominator === 0) {
```

Comment:
Why are we checking for the `numerator` here aswell? 

I think the suggestion would be more robust 
```suggestion
            if (isNaN(this.percentage) && this.denominator === 0) {
```

Reply:
Yes, however, I'm wondering whether you will consider 100/0 also as 0% or as 100%? Because in my eyes, 100 out of 0 seems to be 100%.

Reply:
100/0 is NaN anyways, isn't it?

Reply:
Okay, I do now understand where you come from 0/0 appears to be NaN and 100/0 appears to be Infinity in typescript.

I would then suggest something like 
```
const isNaN = this.numerator === 0 && this.denominator === 0;
if (isNaN(this.percentage || isNaN) {
}
```
So the code tells the reader what shall actually happen (at least to me it is not clear from `this.numerator === 0 && this.denominator === 0` that this is a NaN case and 100/0 is infinity)

I think it would also be good to extract the `if` together with the `else` to a `getUpdatedPercentage` method

The best solution would be to use signals instead and make `percentage` a computed signal, but I would say this is out of scope for the bugfix

Reply:
I am also wondering why we need the NaN check twice 😅

Reply:
```
    ngOnChanges(changes: SimpleChanges) {
        if (changes.percentage) {
            this.percentage = this.getUpdatedPercentage();
            this.chooseProgressBarTextColor();
            this.calculateProgressBarClass();
        }
    }

    private getUpdatedPercentage() {
        const isNaN = this.numerator === 0 && this.denominator === 0;
        if (isNaN(this.percentage) || isNaN) {
            return 0;
        }

        return round(this.percentage);
    }
```

From my perspective this would be a cleaner solution

Reply:
And we should refactor, if possible, this component so that percentage is not passed to it - then we will also only have one NaN check - but not the scope of this PR 👍